libavfilter/vf_drawtext.c

```diff
@@ -1440,10 +1469,13 @@ static int draw_text(AVFilterContext *ctx, AVFrame *frame,
 
     s->var_values[VAR_LINE_H] = s->var_values[VAR_LH] = s->max_glyph_h;
 
-    s->x = s->var_values[VAR_X] = av_expr_eval(s->x_pexpr, s->var_values, &s->prng);
```

Comment:
will it be simpler to add an if the the beginning?
if (s->text_source != ...) {
  unchanged code
}